### ISSUE TITLE
linter: jsx-a11y no-noninteractive-tabindex doesn't work for 0 index when set as string

### ISSUE THREAD

**Issue by MkMan** [oxc-project/oxc#20471]
### What version of Oxlint are you using?

1.56.0

### What command did you run?

`oxlint .`

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
  "$schema": "./node_modules/oxlint/configuration_schema.json",
  "plugins": ["jsx-a11y"],
  "categories": {},
  "rules": {
    "jsx-a11y/no-noninteractive-tabindex": "deny",
  },
  "settings": {},
  "env": {
    "builtin": true
  },
  "globals": {},
  "ignorePatterns": []
}
```

### What happened?

Sample code

```tsx
function App() {
  // run `pnpm lint .` and only the first instance will produce an oxlint error.
  // However, it's not the correct type. React JSX specifies `tabIndex` as being `tabIndex?: number | undefined;`

  return (
    <>
      {/* @ts-expect-error -- tabIndex accepts numbers only, not strings */}
      <div tabIndex="0" /> {/* causes an oxlint error */}
      <div tabIndex={0} /> {/* does NOT cause an oxlint error */}
    </>
  );
}
```

[Reproduction CodeSandbox](https://codesandbox.io/p/devbox/v9krw7?file=%2Fsrc%2FApp.tsx%3A6%2C7)

**Comment by camc314** [comment]
Why are you setting it to the string `0` when it's a type error. What is the use case for this.

Surely the fix here is to set it to the number `0`?

**Comment by MkMan** [comment]
> Why are you setting it to the string `0` when it's a type error. What is the use case for this.
> 
> Surely the fix here is to set it to the number `0`?

Exactly my point @camc314. The rule only throws an error when the value is a string which is a type error and invalid.
When the value is a number the rule doesn't catch that a tab index is set.

This is in violation of the [documentation for the rule](https://oxc.rs/docs/guide/usage/linter/rules/jsx_a11y/no-noninteractive-tabindex.html)

**Comment by camc314** [comment]
Ah I see - thanks, apologies for the confusion

**Comment by MkMan** [comment]
Just updated oxlint and can verify this is fixed. Many thanks @camc314 🙏
